types/repository.d.ts

```diff
@@ -0,0 +1,39 @@
+import type { RepoProvider } from "../utils/providerAPI";
+
+export type RepoIdentifier = { repo_provider: RepoProvider, repo_owner: string, repo_name: string }
+
+export type DbRepo = {
```

Comment:
Why use two types? Both look same to me. Can this be reduced to a single type?

Reply:
Are you talking about `DbRepo` and `DbRepoSerializable` types? 

I have created them both because we can only pass serializable datatypes in props from the `getServerSideProps` function. So one way to reduce the two types into a single type is by removing the `DbRepo` type. But I wanted to retain that because that makes it easier to read an object from the database. For example, I could define `const result: { rows: DbRepo[] }` while making the database query.